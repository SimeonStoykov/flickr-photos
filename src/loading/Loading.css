.loading {
    text-align: center;
    grid-column: 1/-1;
}

.loading-img {
    width: 200px;
}