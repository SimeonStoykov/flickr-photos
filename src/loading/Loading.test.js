import React from 'react';
import { shallow } from 'enzyme';
import '../setupTests';
import Loading from './Loading';

describe('Loading', () => {
    let component;

    beforeEach(() => {
        component = shallow(<Loading />);
    });

    it('should render loading correctly', () => {
        expect(component).toMatchSnapshot();
    });

    it('should contain div with class loading', () => {
        expect(component.find('div.loading')).toBeDefined();
    });

    it('should contain image', () => {
        expect(component.find('img')).toBeDefined();
    });

    it('should contain image with alt attribute', () => {
        expect(component.find('img').prop('alt')).toBeDefined();
    });

    it('should contain div with the text Loading...', () => {
        expect(component.find('div.loading > div').text()).toEqual('Loading...');
    });
});