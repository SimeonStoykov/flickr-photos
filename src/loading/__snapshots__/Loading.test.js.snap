// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render loading correctly 1`] = `
<div
  className="loading"
>
  <img
    alt="Loading..."
    className="loading-img"
    src="test-file-stub"
  />
  <div>
    Loading...
  </div>
</div>
`;
